### ISSUE TITLE
Design Intent - Accessibility Feedback - Appointments Team, Appointments, Oracle Health - Direct Schedule and Request for an Appointment

### ISSUE THREAD

**Issue by briandeconinck** [department-of-veterans-affairs/va.gov-team#88700]
## Next Steps for the VFS team

- [ ] **Assign** this ticket to the team member(s) responsible for addressing feedback provided by Platform.
- [ ] **Comment** on this ticket:
  - If the Platform reviewer has any **Thoughts/Questions** that require responses.
  - When **Must** feedback has been incorporated. As appropriate, link to any other GitHub issues or PRs related to this feedback.
  - When **Should/Consider** feedback has been incorporated, or if any feedback will not be addressed. As appropriate, link to any other GitHub issues or PRs related to this feedback.
- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members`.
- [ ] **Close the ticket** when all feedback has been addressed.

## Thoughts/questions

- This all looks good to me!

## Feedback
Practice areas will document their feedback on the VFS-provided artifacts following the [Must, Should, and Consider Framework](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/must-should-and-consider-framework-for-feedback). Platform Governance reviewers may also provide additional notes that don’t comment on the artifacts themselves but are important for implementation (eg. engineering/coding notes).

- [ ] **Should:** There have been a bunch of products in recent months that have some kind of filter/sort capability, and we've been converging on a more standardized pattern for how those should work (though not yet anything formalized in the design system). This changes something that already exists in production, but I think your updates here could be an opportunity to bring the filter/sort behavior here into alignment with what other products are now doing.
   
   As currently coded (and reflected in your designs), on [Choose a VA location](https://www.figma.com/design/EHR9zBpGnAgc7bHqeIGEoT/OH-Direct-Schedule-MVP-%7C-Appointments-FE?node-id=5092-16966&t=55RpOQqYXSAGfZmy-4) the "Sort facilities" dropdown immediately applies the new sort criteria as soon as the dropdown's selection changes --- and that's all it does. If I'm a screen reader user or a screen magnification user who can't see the list of facilities itself, I don't get any of the visual cues that the list of locations has updated. Instead, I I have to either trust that the change has occurred, or I have the high cognitive load task of comparing what's in the list now with what I remember was their earlier.
   
   In this case, it's not a huge lift to figure it out. But in some other filter/sort scenarios where multiple criteria are applied, it's more challenging to parse out whether anything has happened.
   
   Here's the pattern that other products have adopted and that I would encourage here:
   - Selecting an item from the dropdown doesn't trigger any changes to the page.
   - A button next to/below the dropdown labeled "Sort" (or similar) actually applies the changes when clicked.
   - A summary line of text appears between the sort controls and the radio button options. That text says something like "Showing 5 facilities, sorted by [currently applied criteria]"
   - When a user presses the "Sort" button, keyboard focus moves to the summary line of text.
   
   Here's what that gets you:
   - The summary text gives you a clear statement of what's currently displayed.
   - Moving keyboard focus ensures that immediately after the sort action is applied, screen reader users hear something confirming that the action has occurred.
   - Moving keyboard focus when clicking a button ensures that users don't change context before they're ready. Without the separate button, a user who's arrowing down through the dropdown options may have their focus moved on them unexpectedly, which would be an issue under [WCAG SC 3.2.2](https://www.w3.org/WAI/WCAG22/Understanding/on-input.html).
   
   Hopefully the written explanation makes more sense than whatever I said out loud during the meeting. Please feel free to ping me in Slack if you'd like to discuss this more!

## Governance team actions
- [x] Format feedback as individual tasks (check boxes)
- [ ] Assign this ticket to the VFS team member that opened the Slack request
- [ ] Add the VFS team product label
- [ ] Add the VFS team the feature label (if applicable)
- [x] Add the touchpoint labels
- [x] Add the practice area labels
- [ ] Add the Collaboration Cycle initiative milestone

**Comment by outerpress** [comment]
@briandeconinck what do you think about adding the trigger inside the dropdown list?

My only concern with it being outside the list is that it might not be clear what the state of the list is. For example, if the list says:

`Sort by recent locations [Sort]`

...it might not be clear that I've already sorted the list, since the button will be active, especially if I have memory issues.


<img width="1280" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNmYyQlE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--5f4871228a6fca45d925ab6bbd55dbbece428a36/Small%20Improvements%202024%20_%20Appointments%20FE%20%281%29.png" alt="Small Improvements 2024 _ Appointments FE (1).png" />

**Comment by outerpress** [comment]
@ldelacosta I created design ticket #88860 to handle this

**Comment by outerpress** [comment]
This work is blocked while other teams test this pattern. See [this thread](https://dsva.slack.com/archives/C8E985R32/p1726579725396949).
